### PR TITLE
Init

### PR DESCRIPTION
**Pull request by dmitriyminer**

### CODE REVIEW

File: example-app/cmd/app/main.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		BindDebugPort int    `envconfig:"BIND_DEBUG_PORT"`
+		Consul        struct {
+			Address string `envconfig:"CONSUL_ADDRESS"                   required:"true"`
+		}
+		Zipkin struct {
+			Address      string  `envconfig:"ZIPKIN_ADDRESS"             required:"true"`
+			ReportingAPI string  `envconfig:"ZIPKIN_REPORTING_API"       required:"true"`
+			Fraction     float64 `envconfig:"ZIPKIN_FRACTION"            required:"true"`
+		}
+	}{}
```
Comment (oklymchuk) [inline example-app/cmd/app/main.go:42]: Can take away to type
type AppConfig struct {......

File: example-app/discovery/consul/register.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package consul
+
+import consul "github.com/hashicorp/consul/api"
+
+// ServiceRegister is used to register a new service with the agent
+func ServiceRegister(client *consul.Client, agent *consul.AgentServiceRegistration) error {
+	return client.Agent().ServiceRegister(agent)
+}
```
Comment (oklymchuk) [inline example-app/discovery/consul/register.go:8]: not expedient to use a new function

File: example-app/api/api.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+func API(ver string, promExporter http.Handler) http.Handler {
```
Comment (oklymchuk) [inline example-app/api/api.go:9]: "API" No good name for function

Comment (crnopster) [reply example-app/api/api.go:9]: Func that returns mux.Router must be "newRouter" , "setRoute" etc.

File: example-app/api/example.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opencensus.io/trace"
+	"net/http"
+)
+
+func example(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(
+		r.Context(),
+		"api.example",
```
Comment (crnopster) [inline example-app/api/example.go:14]: Why not to use const here

File: example-app/api/example.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package api
+
+import (
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opencensus.io/trace"
+	"net/http"
+)
```
Comment (crnopster) [inline example-app/api/example.go:9]: Imports must be organized in groups

File: example-app/api/health.go [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package api
+
+import "net/http"
+
+func health(msg string) http.HandlerFunc {
+	body := []byte(msg)
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(body)
```
Comment (crnopster) [inline example-app/api/health.go:10]: What purpose for using underscores

File: example-app/api/health.go [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package api
+
+import "net/http"
+
+func health(msg string) http.HandlerFunc {
```
Comment (crnopster) [inline example-app/api/health.go:5]: Func name must show what that func does

File: example-app/api/version.go [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+func version(v string) http.HandlerFunc {
+	body, _ := json.Marshal(struct {
+		Version string `json:"version"`
+	}{
+		Version: v,
+	})
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(body)
```
Comment (crnopster) [inline example-app/api/version.go:17]: What purpose for using underscores

File: example-app/cmd/app/main.go [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+	openzipkin "github.com/openzipkin/zipkin-go"
+	zipkinHTTPReporter "github.com/openzipkin/zipkin-go/reporter/http"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+var (
+	version = "unknown"
+)
```
Comment (crnopster) [inline example-app/cmd/app/main.go:24]: use const instead of global variables

File: example-app/cmd/app/main.go (hunk 1 above)
Comment (crnopster) [reply example-app/cmd/app/main.go:42]: Better make new file for this struct and add method to make one and return pointer

File: example-app/discovery/consul/register.go (hunk 2 above)
Comment (crnopster) [reply example-app/discovery/consul/register.go:8]: What purpose for using this func?

File: example-app/metrics/exporter.go [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+// NewExporter returns an exporter that exports stats to Prometheus.
+func NewExporter(namespace string) (*prometheus.Exporter, error) {
+	r := stdprometheus.NewRegistry()
+
+	err := r.Register(stdprometheus.NewProcessCollector(stdprometheus.ProcessCollectorOpts{}))
+	if err != nil {
+		return nil, fmt.Errorf("can't register prometheus process collector: %w", err)
```
Comment (crnopster) [inline example-app/metrics/exporter.go:16]: Use const for error message

File: example-app/metrics/exporter.go [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	err = r.Register(stdprometheus.NewGoCollector())
+	if err != nil {
+		return nil, fmt.Errorf("can't register prometheus go collector: %w", err)
```
Comment (crnopster) [inline example-app/metrics/exporter.go:21]: Use const for error message

**Review by radikh — APPROVED** [review]

**Pull request closed without merging**